app/services/foreman_register/registration_url.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module ForemanRegister
+  class RegistrationUrl
+    attr_reader :host
+
+    delegate :registration_token, to: :host
+
+    def initialize(host:)
+      @host = host
+    end
+
+    def url
+      uri = URI.parse(Setting[:foreman_url])
+      uri.scheme = SETTINGS[:require_ssl] ? 'https' : 'http'
```

Comment:
Is this a correct interpretation? `require_ssl` states SSL must be used, but IIRC `false` doesn't mean SSL is not available.